### ISSUE TITLE
[AUTODIFF] [QUESTION]Can pennylane automatic differentiation using parameter shift on real devices?

### ISSUE THREAD

**Issue by YuChenSSR** [PennyLaneAI/pennylane#4631]
Can PennyLane perform automatic differentiation by combining classical differentiation methods on classical computer with parameter shift method on quantum devices?

If possible, could you provide an actual code example to illustrate?

**Comment by CatalinaAlbornoz** [comment]
Hi @YuChenSSR , thank you for your question. The best place to ask these questions is our [discussion Forum](https://discuss.pennylane.ai/) where you can also learn from others' questions.

To answer your question, yes, PennyLane can combine different differentiation methods if needed. For example if you use the PyTorch interface with a hybrid quantum-classical model, then PyTorch will perform the classical differentiation using backpropagation and PennyLane will use parameter-shift to find the gradients over the quantum part.

[This page](https://docs.pennylane.ai/en/stable/introduction/interfaces/torch.html) in the documentation can teach you how to use PennyLane with PyTorch, and this demo on [turning quantum nodes into Torch Layers](https://pennylane.ai/qml/demos/tutorial_qnn_module_torch) has a code example. In that code example all you need to do is change the line that says `dev = qml.device("default.qubit", wires=n_qubits)` to whatever quantum device or simulator you want to use. You can also add the keyword argument `diff_method='parameter-shift'` if you want to use this differentiation method.

[This PennyLane tutorial video](https://www.youtube.com/watch?v=ygzf36vEqkU&list=PLzgi0kRtN5sO8dkomgshjSGDabnjtjBiA&index=10) compares different differentiation methods.

I hope this answers your question!
